### PR TITLE
docs(Table): editable table sample added

### PR DESCRIPTION
**Pull request by MEsteves22**
- Editable table sample added
- Table samples files reorganised and code deduped
- Styling fixes to the `HvTableSection`
- `contentRef` added to `HvSection`
- `container` added to `HvBanner`, `HvSnackbarProvider`, and `HvSnackbar` to set the container the banner/snackbar should slide from

### CODE REVIEW

**Comment by MEsteves22** [comment]
Applitools changes expected

**Comment by plagoa** [comment]
👌 🚀

**Review by plagoa — APPROVED** [review]

**Comment by MEsteves22** [comment]
Introducing some changes to the sample after having some feedback from @VGervasio  🕣

**Comment by MEsteves22** [comment]
Updated ✅

ℹ️ I added the sample to the table section samples. It seemed to make more sense since it follows the new DS styling for tables. Let me know if you have another opinion.

File: packages/core/src/Table/stories/TableSection/TableEditable.tsx [hunk 1]
```diff
@@ -0,0 +1,602 @@
...
+            <div className={edit ? undefined : classes.slide}>
+              <div className={classes.editableActions}>
+                <HvButton
+                  variant="primaryGhost"
+                  onClick={
+                    edit ? () => handleEditRow(row.original.id) : handleAddRow
+                  }
+                >
+                  Save
+                </HvButton>
```
Comment (zettca) [inline packages/core/src/Table/stories/TableSection/TableEditable.tsx:492]: We should use a `form` with a submit button here

**Comment by MEsteves22** [comment]
As suggested by @zettca, I'll be adding a form to the table 🕣

**Review by zettca — COMMENTED** [review]
Sample looks great 👍

Some small notes:

File: packages/core/src/Table/stories/TableHooks/UseHvFilters.tsx [hunk 2]
```diff
@@ -18,27 +16,14 @@ import {
...
-  </HvTableRow>
-);
+import {
+  makeData,
+  getColumns,
+  AssetEvent,
+  range,
+  EmptyRow,
+  NoDataRow,
+} from "../storiesUtils";
```
[the remark this answers is not in the document]
Comment (MEsteves22) [reply packages/core/src/Table/stories/TableHooks/UseHvFilters.tsx:26]: I'll add back the `NoDataRow` and `EmptyRow` for now. We can discuss later if we should promote them.

File: packages/core/src/Table/stories/TableSection/TableEditable.tsx [hunk 3]
```diff
@@ -0,0 +1,602 @@
...
+            startIcon={<Add />}
+            variant="primaryGhost"
+            onClick={() => setAdd(true)}
+          >
+            Add Row
+          </HvButton>
+        }
+        contentRef={setContentRef}
+      >
+        <Main add={add} setAdd={(value: boolean) => setAdd(value)} />
```
[the remark this answers is not in the document]
Comment (MEsteves22) [reply packages/core/src/Table/stories/TableSection/TableEditable.tsx:598]: Yup, I'll change the name

File: packages/core/src/Table/stories/TableSection/TableEditable.tsx [hunk 4]
```diff
@@ -0,0 +1,602 @@
...
+
+  const renderEditableRow = (row?: HvRowInstance<AssetEvent, string>) => {
+    const cols = columns.filter((col) => col.variant !== "actions");
+    const edit = !!row;
+
+    return (
+      <Fragment key={`editable_row_group_${row?.original.id}`}>
+        <HvTableRow
+          key="editable_row"
+          role="row"
```
[the remark this answers is not in the document]
Comment (MEsteves22) [reply packages/core/src/Table/stories/TableSection/TableEditable.tsx]: I'll remove the unnecessary roles and keys

File: packages/core/src/Table/stories/TableSection/TableEditable.tsx [hunk 5]
```diff
@@ -0,0 +1,602 @@
...
+};
+
+const Main = ({
+  add,
+  setAdd,
+}: {
+  add: boolean;
+  setAdd: (value: boolean) => void;
+}) => {
+  const { data, loading, fetchData, pageCount, mutateData } = useServerData();
```
[the remark this answers is not in the document]
Comment (MEsteves22) [reply packages/core/src/Table/stories/TableSection/TableEditable.tsx:126]: Good idea, I'll do that

File: packages/core/src/Table/stories/TableSection/TableEditable.tsx (hunk 1 above)
Comment (MEsteves22) [reply packages/core/src/Table/stories/TableSection/TableEditable.tsx:493]: Yes, I'll add the `form`

**Comment by MEsteves22** [comment]
Updated with comments ✅

**Review by plagoa — APPROVED** [review]

**Pull request merged**